islands/VoteButton.tsx

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import type { Item, Vote } from "@/utils/db.ts";
+import { useState } from "preact/hooks";
+
+export default function VoteButton(
+  props: { item: Item; votes: Vote[] | [] },
+) {
```

Comment:
Could you please create a `VoteButtonProps` interface for these properties? I'd also like there to be an `isVoted` property, and it be calculated within the page handler.